File: Coursework/Level 6/Implementation Task/Patient Access Control.py
def check_access(role):
    if role != "Doctor":
        raise PermissionError("Access Denied: Only Doctors can view patient records.")
    return "Access Granted: Patient Records Loading..."

# Execution
try:
    current_role = input("Enter your role: ")
    print(check_access(current_role))
except PermissionError as e:
    print(f"Security Alert: {e}")

File: Coursework/Level 6/Implementation Task/Patient Access Control.java
import java.util.Scanner;

public class HospitalSystem {
    public static void checkAccess(String role) throws Exception {
        if (!role.equals("Doctor")) {
            throw new Exception("Unauthorized Access: Role must be Doctor.");
        }
        System.out.println("Access Granted.");
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter role: ");
        String userRole = scanner.nextLine();

        try {
            checkAccess(userRole);
        } catch (Exception e) {
            System.out.println("Error: " + e.getMessage());
        }
    }
}